var/spack/repos/builtin/packages/libsonata/package.py

```diff
@@ -26,8 +27,8 @@ class Libsonata(CMakePackage):
     depends_on('cmake@3.3:', type='build')
     depends_on('py-setuptools-scm', type='build', when='@0.1:')
     depends_on('fmt@4.0:')
-    depends_on('highfive+mpi', when='+mpi')
-    depends_on('highfive~mpi', when='~mpi')
+    depends_on('highfive+mpi@2.2.1', when='+mpi')
+    depends_on('highfive~mpi@2.2.1', when='~mpi')
```

Comment:
Can we make this version dependent? Pinning it like this is dangerous.

Reply:
Sure you are right. 

So you mean to : 
```
    depends_on('highfive+mpi@2.2.1:', when='+mpi')
    depends_on('highfive~mpi@2.2.1:', when='~mpi')
```
or something more fancy ?

Reply:
like :

```
depends_on('highfive+mpi', when='@:0.1.2+mpi')
depends_on('highfive~mpi', when='@:0.1.2~mpi')
depends_on('highfive+mpi@2.2.1:', when='@0.1.3:+mpi')
depends_on('highfive~mpi@2.2.1:', when=' @0.1.3:~mpi')
```

Reply:
The latter. Given the error message, it looks like there needs to be a version pin somewhere else, too…

Reply:
Yeah everything is really entangled ...
Ok thx ! I add this and I will bump the version of the nse  packages also (we have new releases for bluepy and bluepysnap)